osc_preferences.h

```diff
@@ -0,0 +1,126 @@
+#ifndef __OSC_PREFERENCES_H__
+#define __OSC_PREFERENCES_H__
+
+#include <glib/glist.h>
+#include <glib/gprintf.h>
+
+typedef struct osc_preferences OscPreferences;
+typedef struct oscplot_preferences OscPlotPreferences;
+typedef struct iio_device_preferences IioDevPreferences;
+typedef struct iio_channel_preferences IioChnPreferences;
+
+struct oscplot_preferences {
+        unsigned int *sample_count;
+        GList *devices_pref_list;
+};
+
+struct iio_channel_preferences {
+        char *chn_name;
+        unsigned int *constraints;
+};
+
+struct iio_device_preferences {
+        char *dev_name;
+        double *fft_correction;
+        GList *channels_pref_list;
+};
+
+struct osc_preferences {
+        struct oscplot_preferences *plot_preferences;
+};
+
+
+/* Construct/Destruct methods */
+
+
+/* IioChnPreferences */
+
+static inline IioChnPreferences* iio_chn_preferences_new(const char *chn_name)
+{
+        IioChnPreferences *chn_pref = g_new(IioChnPreferences, 1);
+
+        chn_pref->chn_name = g_strdup(chn_name);
+        chn_pref->constraints = NULL;
+
+        return chn_pref;
+}
+
+static inline void iio_chn_preferences_free(IioChnPreferences *pref)
+{
+        if (pref->chn_name) {
+                g_free(pref->chn_name);
```

Comment:
curios: is `g_free()` NULL tolerant?
so that you can skip the null check

Reply:
It is NULL tolerant. Thanks!